### PR TITLE
Improved attribute handling with types and Attribute support

### PR DESCRIPTION
**Pull request by stefanScrumble**
## Summary

Properly handle attributes with type templates by parsing the phpdoc and outputting the the ide-helper file with type templates included as FQN.

Handle the Laravel Attribute return type for attribute getters and setters.

## Type of change

- [X ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)

### Checklist
- [X ] Existing tests have been adapted and/or new tests have been added
- [ X] Add a CHANGELOG.md entry
- [ X] Update the README.md
- [ X] Code style has been fixed via `composer fix-style`

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "barryvdh/laravel-ide-helper",
+    "name": "scrumble-nl/laravel-ide-helper",
```
Comment (barryvdh) [inline composer.json:2]: Can you change this back?

**Comment by barryvdh** [comment]
@mfn what do you think?

File: composer.json (hunk 1 above)
Comment (stefanScrumble) [reply composer.json]: Yes, done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -80,6 +80,17 @@ All notable changes to this project will be documented in this file.
 - Add support for `immutable_date:*` and `immutable_datetime:*` casts. [#1380 / thekonz](https://github.com/barryvdh/laravel-ide-helper/pull/1380)
 - Add support for attribute accessors marked as protected. [#1339 / pindab0ter](https://github.com/barryvdh/laravel-ide-helper/pull/1339)
 
+2024-01-04, v2.14.0
+------------------
```
Comment (barryvdh) [inline CHANGELOG.md:84]: This would go in 3.1

File: src/helpers/PhpDocTypeParser.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Barryvdh\LaravelIdeHelper\helpers;
```
Comment (barryvdh) [inline src/helpers/PhpDocTypeParser.php:5]: Not sure if 'helpers' is appropriate, perhaps something more descriptive like Parsers? But it should follow PSR12 and use StudlyCaps

Comment (stefanScrumble) [reply src/helpers/PhpDocTypeParser.php]: Makes sense, fixed! And I Improved the filename to more accurately reflect it's use.

**Review by barryvdh — APPROVED** [review]

**Pull request merged**